packages/devtools/lib/src/tables.dart

```diff
@@ -556,13 +686,92 @@ abstract class Column<T> {
   String toString() => title;
 
   void renderToElement(CoreElement cell, T dataObject) {
-    final String content = render(getValue(dataObject));
+    final String content = render(getDisplayValue(dataObject));
     if (usesHtml) {
       cell.setInnerHtml(content);
     } else {
       cell.text = content;
     }
+    cell.tooltip = getTooltip(dataObject);
   }
+
+  String _getAlignmentCss(ColumnAlignment alignment) {
+    switch (alignment) {
+      case ColumnAlignment.left:
+        return 'left';
+      case ColumnAlignment.right:
+        return 'right';
+      case ColumnAlignment.center:
+        return 'center';
+      default:
+        throw Exception('Invalid column alignment: $alignment');
+    }
+  }
+}
+
+abstract class TreeColumn<T extends TreeNode> extends Column<T> {
+  TreeColumn(
+    title, {
+    int fixedWidthPx,
+    double fractionWidth,
+  }) : super(title, fixedWidthPx: fixedWidthPx, fractionWidth: fractionWidth);
+
+  static const treeToggleWidth = 14;
+
+  final StreamController<T> _nodeExpandedController =
+      StreamController<T>.broadcast();
+
+  Stream<T> get onNodeExpanded => _nodeExpandedController.stream;
+
+  final StreamController<T> _nodeCollapsedController =
+      StreamController<T>.broadcast();
+
+  Stream<T> get onNodeCollapsed => _nodeCollapsedController.stream;
+
+  int _getNodeIndentPx(T dataObject) {
+    int indentWidth = dataObject.level * treeToggleWidth;
+    if (!dataObject.isExpandable) {
+      // If the object is not expandable, we need to increase the width of our
+      // spacer to account for the missing tree toggle.
+      indentWidth += TreeColumn.treeToggleWidth;
```

Comment:
expandCollapseWidth?

Reply:
I think treeToggleWidth is a more fitting name because it is consistent with the class name for the custom element - TreeToggle.